### PR TITLE
Background color property

### PR DESCRIPTION
**Pull request by georonathan47**
Refactored the code to suit the minimal diff... The entire bar is now wrapped around `ColoredBox` which defaults to `Theme.of(context).bottomAppBarColor`. The requirements to add a background is no longer compulsory like it was in the closed PR

### CODE REVIEW

File: example/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>8.0</string>
+  <string>11.0</string>
```
Comment (lukepighetti) [inline example/ios/Flutter/AppFrameworkInfo.plist]: why was this changed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## [3.3.2] - Dec 27, 2022
+
+- Added option to choose your own background color (Default is still white 😄).
+
```
Comment (lukepighetti) [inline CHANGELOG.md]: You can remove this, I'll do the notes and date when I publish the releases

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -127,7 +127,7 @@
 		97C146E61CF9000F007C117D /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1020;
+				LastUpgradeCheck = 1300;
```
Comment (lukepighetti) [inline example/ios/Runner.xcodeproj/project.pbxproj]: why was this changed?

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -272,7 +272,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
Comment (lukepighetti) [inline example/ios/Runner.xcodeproj/project.pbxproj:275]: why was this changed?

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -350,7 +350,7 @@
...
```
Comment (lukepighetti) [inline example/ios/Runner.xcodeproj/project.pbxproj:353]: why was this changed?

File: example/ios/Runner.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -399,7 +399,7 @@
...
```
Comment (lukepighetti) [inline example/ios/Runner.xcodeproj/project.pbxproj]: why was this changed?

File: example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1020"
+   LastUpgradeVersion = "1300"
```
Comment (lukepighetti) [inline example/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme]: why was this changed?

File: example/ios/Runner/Info.plist [hunk 8]
```diff
@@ -41,5 +41,7 @@
 	</array>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<false/>
+	<key>CADisableMinimumFrameDurationOnPhone</key>
+	<true/>
```
Comment (lukepighetti) [inline example/ios/Runner/Info.plist:45]: why was this changed?

File: lib/salomon_bottom_bar.dart [hunk 9]
```diff
@@ -56,103 +60,87 @@ class SalomonBottomBar extends StatelessWidget {
...
-                              : EdgeInsets.only(left: itemPadding.left * t)),
-                      child: Row(
-                        children: [
-                          IconTheme(
-                            data: IconThemeData(
-                              color: Color.lerp(
-                                  _unselectedColor, _selectedColor, t),
-                              size: 24,
+    return ColoredBox(
+    color: backgroundColor ?? theme.bottomAppBarColor,
```
Comment (lukepighetti) [inline lib/salomon_bottom_bar.dart]: check your formatting

File: example/pubspec.lock [hunk 10]
```diff
@@ -73,28 +66,28 @@ packages:
...
-    version: "0.1.3"
+    version: "0.1.5"
   meta:
     dependency: transitive
     description:
       name: meta
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "1.7.0"
+    version: "1.8.0"
```
Comment (lukepighetti) [inline example/pubspec.lock]: please don't change pubspec.lock

File: example/lib/main.dart [hunk 11]
```diff
@@ -24,10 +24,12 @@ class _MyAppState extends State<MyApp> {
         visualDensity: VisualDensity.adaptivePlatformDensity,
       ),
       home: Scaffold(
+        backgroundColor: Colors.grey,
```
Comment (lukepighetti) [inline example/lib/main.dart]: can we showcase the adjustable background color?

File: pubspec.lock [hunk 12]
```diff
@@ -21,35 +21,28 @@ packages:
...
     source: hosted
-    version: "1.2.0"
-  charcode:
-    dependency: transitive
-    description:
-      name: charcode
-      url: "https://pub.dartlang.org"
-    source: hosted
-    version: "1.3.1"
+    version: "1.2.1"
```
Comment (lukepighetti) [inline pubspec.lock]: let's not change pubspec.lock

File: example/lib/main.dart [hunk 13]
```diff
@@ -24,10 +24,12 @@ class _MyAppState extends State<MyApp> {
...
         appBar: AppBar(
           title: Text(MyApp.title),
         ),
         bottomNavigationBar: SalomonBottomBar(
+          // backgroundColor: Colors.white,
```
Comment (georonathan47) [inline example/lib/main.dart]: This is where we can showcase the adjustable background color

File: example/lib/main.dart (hunk 11 above)
Comment (georonathan47) [reply example/lib/main.dart]: This backround color just shows the background color of the page selected

File: CHANGELOG.md (hunk 2 above)
Comment (georonathan47) [reply CHANGELOG.md]: ok

File: example/lib/main.dart (hunk 13 above)
Comment (georonathan47) [reply example/lib/main.dart]: The background color defaults to theme.bottomAppBarColor when no backgroundColor property is specified

File: example/ios/Flutter/AppFrameworkInfo.plist (hunk 1 above)
Comment (georonathan47) [reply example/ios/Flutter/AppFrameworkInfo.plist]: my flutter default configs

File: pubspec.lock (hunk 12 above)
Comment (georonathan47) [reply pubspec.lock]: I am not manually changing it. I think it's got to do with my defualt flutter settings

**Comment by lukepighetti** [comment]
There are still a lot of changes that are unnecessary, we'll have to remove them before this can be merged. There should only be changes in the `example/lib/main.dart` and in `lib/salomon_bottom_bar.dart`

**Comment by lukepighetti** [comment]
Closing due to being stale. Happy to merge this feature but it has to meet the criteria laid out above. Minimal diff, no other changes. I will handle the rest.

**Comment by georonathan47** [comment]
I’ve had d some deployments to handle off late which explains the PR being staleI’d reopen it and add just the colored box as the minimal diffSent from my iPhoneOn 14 Mar 2023, at 14:56, Luke Pighetti ***@***.***> wrote:﻿
Closed #38.

**Comment by lukepighetti** [comment]
Closed by #44
